powerinfer-py/powerinfer/solver.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+# coding=utf-8
+import argparse
+from cvxopt.glpk import ilp
+import numpy as np
+from cvxopt import matrix
+import torch
+import pickle
+
+def solve_gpu_split(
+    activation_path: str,
+    neuron: int,
+    capacity: int,
+    layer: int,
+    batch: int,
+    threshold: int,
+):
+    # Processing activation data
+    values = []
+    for i in range(layer):
+        # Load and sort activation data for each layer
+        freq = torch.load(f"{activation_path}/activation_{i}.pt")
+        freq, _ = torch.sort(freq, descending=True)
+        freq = freq * -1.0
+        freq = freq.view(-1, batch)
+        freq = freq.sum(dim=1)
+        freq = freq.tolist()
+        values += freq
+
+    # Padding zero values for additional constraints
+    for i in range(layer):
+        values += [0.0]
+    c = np.array(values, dtype=float)
+    c = matrix(c)
+
+    # Setting capacity and neuron count per batch
+    CAP = capacity
+    CAP = int(CAP / batch)
+    neuron = int(neuron / batch)
+    coeff = []
+    h = []
+
+    # Constraint 1: Total neuron activation constraint
```

Comment:
Three constraints: 
1. Try to minimize total neural activations.
2. Threshold constraint for GPU split per layer
3. Upper bound on neuron activations